### PR TITLE
Add Third Reality plug, motion, water leak sensor

### PR DESCRIPTION
**Pull request by hwzolin**
## Proposed change

This pull request includes the addition of custom clusters for Third Reality devices in the `zhaquirks` library. The changes involve defining new clusters for motion sensors, plugs, and water leak sensors, and registering these clusters with specific attributes and behaviors.

New custom clusters and attributes:

* [`zhaquirks/thirdreality/motion_sensor.py`](diffhunk://#diff-c8261beab8aa694a2e249098b4fcf4c1d0922693273f0c9cc4b6ce57a1717352R1-R37): Added `ThirdRealityMotionCluster` with `cooldown_time` attribute and registered the cluster.
* [`zhaquirks/thirdreality/plug.py`](diffhunk://#diff-d679d814b3e7c4bd6bdc88fb0e3c9039f82c65fcef23748bda30197ca74dac89R1-R38): Added `ThirdRealityPlugCluster` with `reset_summation_delivered` attribute and registered the cluster with additional device applicability.
* [`zhaquirks/thirdreality/waterLeak_sensor.py`](diffhunk://#diff-b02207b2ed9378489ea1ab83bbeee5a95dd706797be4a6d1c5d3309db81bee6cR1-R48): Added `ThirdRealityWaterLeakCluster` with `enable_siren` and `siren_time` attributes and registered the cluster.

## Checklist

- [ ] The changes are tested and work correctly
- [ ] `pre-commit` checks pass / the code has been formatted using Black
- [ ] Tests have been added to verify that the new code works

### CODE REVIEW

**Comment by hwzolin** [comment]
@puddly Hello, this is our company’s private Cluster support for the Plug for quirks v2. I have verified locally that the function is normal. Please take a look. If there is no problem, please help merge it. Thank you!

**Comment by hwzolin** [comment]
@puddly @TheJulianJES 
Hello, I have made some modifications to the original plug code, and added quirks v2 support code for the private cluster of the water leak sensor and motion sensor devices.
Please check it out, thank you!

File: zhaquirks/thirdreality/motion_sensor.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+(
+    QuirkBuilder("Third Reality, Inc", "3RMS16BZ")
+    .replaces(ThirdRealityMotionCluster)
+    .number(
+        attribute_name=ThirdRealityMotionCluster.AttributeDefs.cooldown_time.name,
+        min_value=5,
+        max_value=3600,
+        cluster_id=ThirdRealityMotionCluster.cluster_id,
+    )
```
Comment (TheJulianJES) [inline zhaquirks/thirdreality/motion_sensor.py]: We now need a translation key and fallback name (that'll be the name used by the entity in HA, although it'll be translatable by the translation key – fallback name is just the English default.)
```suggestion
    .number(
        attribute_name=ThirdRealityMotionCluster.AttributeDefs.cooldown_time.name,
        min_value=5,
        max_value=3600,
        cluster_id=ThirdRealityMotionCluster.cluster_id,
        translation_key="cooldown_time",
        fallback_name="Cooldown time",
    )

```

File: zhaquirks/thirdreality/plug.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+(
+    QuirkBuilder("Third Reality, Inc", "3RSP02028BZ")
+    .also_applies_to("Third Reality, Inc", "3RSPE01044BZ")
+    .replaces(ThirdRealityPlugCluster)
+    .write_attr_button(
+        attribute_name=ThirdRealityPlugCluster.AttributeDefs.reset_summation_delivered.name,
+        attribute_value=0x01,
+        cluster_id=ThirdRealityPlugCluster.cluster_id,
+        translation_key="reset_summation_delivered",
```
Comment (TheJulianJES) [inline zhaquirks/thirdreality/plug.py:34]: We also need a fallback name here:
```suggestion
    .replaces(ThirdRealityPlugCluster)
    .write_attr_button(
        attribute_name=ThirdRealityPlugCluster.AttributeDefs.reset_summation_delivered.name,
        attribute_value=0x01,
        cluster_id=ThirdRealityPlugCluster.cluster_id,
        translation_key="reset_summation_delivered",
        fallback_name="Reset summation delivered",
```

File: zhaquirks/thirdreality/waterLeak_sensor.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        )
+
+
+(
+    QuirkBuilder("Third Reality, Inc", "3RWS18BZ")
+    .replaces(ThirdRealityWaterLeakCluster)
+    .switch(
+        attribute_name=ThirdRealityWaterLeakCluster.AttributeDefs.enable_siren.name,
+        cluster_id=ThirdRealityWaterLeakCluster.cluster_id,
+    )
```
Comment (TheJulianJES) [inline zhaquirks/thirdreality/waterLeak_sensor.py]: We also need a translation key and fallback name here:
```suggestion
    .switch(
        attribute_name=ThirdRealityWaterLeakCluster.AttributeDefs.enable_siren.name,
        cluster_id=ThirdRealityWaterLeakCluster.cluster_id,
        translation_key="enable_siren",
        fallback_name="Enable siren",
    )
```

File: zhaquirks/thirdreality/waterLeak_sensor.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    .number(
+        attribute_name=ThirdRealityWaterLeakCluster.AttributeDefs.siren_time.name,
+        min_value=0,
+        max_value=255,
+        cluster_id=ThirdRealityWaterLeakCluster.cluster_id,
+    )
```
Comment (TheJulianJES) [inline zhaquirks/thirdreality/waterLeak_sensor.py]: We also need a translation key and fallback name here:
```suggestion
    .number(
        attribute_name=ThirdRealityWaterLeakCluster.AttributeDefs.siren_time.name,
        min_value=0,
        max_value=255,
        cluster_id=ThirdRealityWaterLeakCluster.cluster_id,
        translation_key="siren_time",
        fallback_name="Siren time",
    )
```

File: zhaquirks/thirdreality/motion_sensor.py [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    cluster_id = 0xFF01
+
+    class AttributeDefs(BaseAttributeDefs):
+        """Define the attributes of a private cluster."""
+
+        cooldown_time: Final = ZCLAttributeDef(
+            id=0x0001,
+            type=t.uint16_t,
+            is_manufacturer_specific=True,
+            name="Cooldown_Time",
```
Comment (TheJulianJES) [inline zhaquirks/thirdreality/motion_sensor.py]: The `name` is already defined by the attribute name when using this definition style, we should remove it.
```suggestion
```

File: zhaquirks/thirdreality/plug.py [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    cluster_id = 0xFF03
+
+    class AttributeDefs(BaseAttributeDefs):
+        """Define the attributes of a private cluster."""
+
+        reset_summation_delivered: Final = ZCLAttributeDef(
+            id=0x0000,
+            type=t.uint8_t,
+            is_manufacturer_specific=True,
+            name="Reset_summation_delivered",
```
Comment (TheJulianJES) [inline zhaquirks/thirdreality/plug.py]: The `name` is already defined by the attribute name when using this definition style, we should remove it.
```suggestion
```

File: zhaquirks/thirdreality/waterLeak_sensor.py [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+    cluster_id = 0xFF01
+
+    class AttributeDefs(BaseAttributeDefs):
+        """Define the attributes of a private cluster."""
+
+        enable_siren: Final = ZCLAttributeDef(
+            id=0x0010,
+            type=t.uint8_t,
+            is_manufacturer_specific=True,
+            name="Enable_or_Disable_siren",
```
Comment (TheJulianJES) [inline zhaquirks/thirdreality/waterLeak_sensor.py]: The `name` is already defined by the attribute name when using this definition style, we should remove it.
```suggestion
```

File: zhaquirks/thirdreality/waterLeak_sensor.py [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+        )
+
+        siren_time: Final = ZCLAttributeDef(
+            id=0x0011,
+            type=t.uint8_t,
+            is_manufacturer_specific=True,
+            name="Siren_timer",
```
Comment (TheJulianJES) [inline zhaquirks/thirdreality/waterLeak_sensor.py]: The `name` is already defined by the attribute name when using this definition style, we should remove it.
```suggestion
```

File: zhaquirks/thirdreality/motion_sensor.py [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+        )
+
+
+(
+    QuirkBuilder("Third Reality, Inc", "3RMS16BZ")
+    .replaces(ThirdRealityMotionCluster)
+    .number(
+        attribute_name=ThirdRealityMotionCluster.AttributeDefs.cooldown_time.name,
+        min_value=5,
+        max_value=3600,
```
Comment (TheJulianJES) [inline zhaquirks/thirdreality/motion_sensor.py]: We should add the unit here (and import `UnitOfTime`) .
I assume it's in seconds? Then do it like this:
```suggestion
        max_value=3600,
        unit=UnitOfTime.SECONDS,
```
(and add `from zigpy.quirks.v2.homeassistant import UnitOfTime` add the top for the import)

File: zhaquirks/thirdreality/waterLeak_sensor.py [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+    .switch(
+        attribute_name=ThirdRealityWaterLeakCluster.AttributeDefs.enable_siren.name,
+        cluster_id=ThirdRealityWaterLeakCluster.cluster_id,
+        translation_key="enable_siren",
+        fallback_name="Enable siren",
+    )
+    .number(
+        attribute_name=ThirdRealityWaterLeakCluster.AttributeDefs.siren_time.name,
+        min_value=0,
+        max_value=255,
```
Comment (TheJulianJES) [inline zhaquirks/thirdreality/waterLeak_sensor.py]: We should add the unit here (and import `UnitOfTime`) .
I assume it's in seconds? Then do it like this:
```suggestion
        max_value=255,
        unit=UnitOfTime.SECONDS,
```
If it's minutes, replace with `MINUTES`.
(and add `from zigpy.quirks.v2.homeassistant import UnitOfTime` add the top for the import)

**Review by TheJulianJES — COMMENTED** [review]
The `waterLeak_sensor.py` file also needs to be renamed to something like `water_leak_sensor.py`.

**Comment by TheJulianJES** [comment]
Why did you close this? Do not open another PR for this please. Just commit changes to your `dev` branch. They'll appear here.
If you have issues, let me know and I'll be able to commit the few last changes. I just need to know if the config entities are of unit "seconds" or "minutes".
An import is just needed for the suggestion to work.

**Comment by hwzolin** [comment]
I'm sorry to trouble you. I originally thought that I needed to modify the code myself and then resubmit it.
May I ask what I should do now? Pull down the latest code first, and then, according to your requirements, modify it and resubmit it?

**Comment by TheJulianJES** [comment]
I've pushed the changes to your `dev` branch. You could just update your local Git repo with the changes I pushed, commit new ones, and then push those.

Otherwise, I can also finish this PR, as it's almost done.
Please tell me the unit for the "Cooldown time" of the motion sensor. Minutes or seconds?
And what is the unit for the "Siren time" of the water leak sensor. Minutes or seconds?

I can have the PR updated and merged in a few minutes then.

**Comment by hwzolin** [comment]
@TheJulianJES  
OK, thanks for your help.
The unit for the "Cooldown time" of the motion sensor is seconds. 
The unit for the "Siren time" of the water leak sensor is minutes.

**Comment by TheJulianJES** [comment]
Ok, thanks. I'll get the PR updated.

**Comment by TheJulianJES** [comment]
Ok, I think this should be good to go now.
I've also renamed `cooldown_time` to `detection_interval` for the motion sensor (entity). It's just to match what we already have in ZHA for other devices.

**Review by TheJulianJES — APPROVED** [review]
Thanks for the PR!

**Pull request merged**